plugins/github-auth-plugin/src/github-auth-plugin.ts

```diff
@@ -11,22 +11,54 @@
 import * as theia from '@theia/plugin';
 import * as che from '@eclipse-che/plugin';
 
-export function start(context: theia.PluginContext): void {
+export async function start(context: theia.PluginContext): Promise<void> {
     if (theia.plugins.getPlugin('github.vscode-pull-request-github')) {
-        const command = {
-            id: 'github-plugin-authenticate',
-            label: 'GitHub authenticate'
-        };
-        context.subscriptions.push(theia.commands.registerCommand(command, async () => {
-            const token = await che.github.getToken();
-            const conf = theia.workspace.getConfiguration();
-            await conf.update('githubPullRequests.hosts', [{
-                host: 'github.com',
-                token
-            }], theia.ConfigurationTarget.Global);
-            theia.window.showWarningMessage('GitHub token has been set to preferences. ' +
-                'Refresh the page to reinitialise the vscode GitHub pull-request plugin with the token');
-        }));
+        let session: theia.AuthenticationSession | undefined = context.workspaceState.get('session');
+        const onDidChangeSessions = new theia.EventEmitter<theia.AuthenticationProviderAuthenticationSessionsChangeEvent>();
+        theia.authentication.registerAuthenticationProvider({
+                id: 'github',
+                label: 'GitHub',
+                supportsMultipleAccounts: false,
+                onDidChangeSessions: onDidChangeSessions.event,
+                getSessions: async () => {
+                    if (session) {
+                        return [session];
+                    } else {
+                        return [];
+                    }
+                },
+                login: async (scopes: string[]) => {
+                    const githubUser = await che.github.getUser();
+                    session = {
+                        id: 'github-session',
+                        accessToken: await che.github.getToken(),
```

Comment:
what happen on fresh install, I'm not sure che.github.getToken() or che.github.getUser() is returning something ?

Reply:
In this case the GitHub authentication page appears:
https://github.com/eclipse/che-theia/blob/2f11124e7de0e50feed25fddf91a9058c37a2a1b/extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts#L101-L117
Then if user authenticates it the GitHub token is setup and these functions work as expected.